### DISCUSSION TITLE
Form component by TypeScript does not support defaultValues

### DISCUSSION THREAD

**Discussion by nimabrownlee** [react-hook-form/react-hook-form#5463] [Q&A] [answered]
Hey, I used the exact same Form component found here: https://github.com/react-hook-form/react-hook-form/blob/master/examples/V7/typescript/UseFormMethods.tsx

```
const methods = useForm<TFormValues>({
    defaultValues: {
      name: 'hi'
    }
  })
```

Typescript complains and says that I have to implement all members of `useForm()`.
Its because of this part: const Form = <TFormValues extends Record<string, any> = Record<string, any>>({
But I don't know enough TS to know what I have to change.

Here is a codesandbox: https://codesandbox.io/s/eloquent-gareth-osdwx?file=/src/App.tsx

and error message I get: https://i.imgur.com/YXMUd3S.png

**Comment by kotarella1110** [comment, marked as answer]
@nimabrownlee Because `Form` component is generics (`TFormValues`), since the specific schema of `FormValues` is unknown at that point.
This can be solved by adding `defaultValues` prop to the `Form` component.

https://codesandbox.io/s/5463-imtgv
